def personal_sum(numbers):
    result = 0
    incorrect_data = 0
    for i in numbers:
        try:
            result += i
            #print(result)
        except TypeError as exc:
            incorrect_data += 1
            print(f'Некорекктный тип данных для подсчёта суммы - {i})')
    return result, incorrect_data

# numbers = "1, 2, 3"
# print(personal_sum(numbers))
def calculate_average(numbers):
    try:
        result_1 = personal_sum(numbers)
        return result_1[0] / (len(numbers) - result_1[1])
    except ZeroDivisionError as exc:
        return 0
    except TypeError as exc:
        return print(f'В numbers записан некорректный тип данных')
    if isinstance(numbers, int):
        return None


print(f'Результат 1: {calculate_average("1, 2, 3")}')
print(f'Результат 2: {calculate_average([1, "Строка", 3, "Ещё Строка"])}')
print(f'Результат 3: {calculate_average(567)}')
print(f'Результат 4: {calculate_average([42, 15, 36, 13])}')



Консоль:
C:\Users\Дом\PycharmProjects\djangoProject\venv\Scripts\python.exe C:/Users/Дом/PycharmProjects/djangoProject/Задания/module_8_2.py
Некорекктный тип данных для подсчёта суммы - 1)
Некорекктный тип данных для подсчёта суммы - ,)
Некорекктный тип данных для подсчёта суммы -  )
Некорекктный тип данных для подсчёта суммы - 2)
Некорекктный тип данных для подсчёта суммы - ,)
Некорекктный тип данных для подсчёта суммы -  )
Некорекктный тип данных для подсчёта суммы - 3)
Результат 1: 0
Некорекктный тип данных для подсчёта суммы - Строка)
Некорекктный тип данных для подсчёта суммы - Ещё Строка)
Результат 2: 2.0
В numbers записан некорректный тип данных
Результат 3: None
Результат 4: 26.5

Process finished with exit code 0
